autoprecompiles/src/lib.rs

```diff
@@ -510,11 +524,34 @@ pub fn exec_receive<T: FieldElement>(machine: &SymbolicMachine<T>) -> SymbolicBu
     r
 }
 
+// Check that the number of register memory bus interactions in the precompile is even.
+// The check is performed on the total number and per address.
+pub fn check_precompile<T: FieldElement>(machine: &SymbolicMachine<T>) -> bool {
+    let (total_count, count_per_addr) = machine
+        .bus_interactions
+        .iter()
+        .filter_map(|bus_int| bus_int.clone().try_into().ok())
+        .filter(|mem_int: &MemoryBusInteraction<T>| matches!(mem_int.ty, MemoryType::Register))
+        .map(|mem_int| {
+            mem_int.try_addr_u32().unwrap_or_else(|| {
+                panic!(
+                    "Register memory access must have constant address but found {}",
+                    mem_int.addr
+                )
+            })
+        })
+        .fold((0, BTreeMap::new()), |(count, mut map), addr| {
+            *map.entry(addr).or_insert(0) += 1;
+            (count + 1, map)
+        });
+
+    total_count % 2 == 0 && count_per_addr.values().all(|&v| v % 2 == 0)
```

Comment:
doesn't the latter imply the former? also, the docs seem incorrect here, as we do more than check the number of register bus interactions is even, we check evenness for each address.

Reply:
it does  assuming that the assertion before it catches the non-constant addresses, just added for extra safety